docs/fundamentals/k8s.md

```diff
@@ -0,0 +1,159 @@
+
+## Automatically translate a Flow to Kubernetes concept
+
+
+```{hint}
+Manually building these Kubernetes YAML object is long and cumbersome. Therefore we provide a helper function {meth}`~jina.Flow.to_kubernetes_yaml` that does most of this
+translation work automatically. 
+```
+
+This helper function can be called from:
+* our Python interface to translate a Flow define in Python to k8s yaml files
+* our CLI interface to export a YAML Flow to k8s yaml files
+
+```{seealso}
+More detail in the {ref}`Flow export documentation <kubernetes-export>`
+```
+
+
+## Extra Kubernetes options
+
+In general, Jina follows the following principle when it comes to deploying in Kubernetes:
+You, the user, know your use case and requirements the best.
+This means that while Jina generates configurations for you that run out of the box, as a professional user you should
+always see them as just a starting point to get you off the ground.
+
+```{hint}
+The export funciton{meth}`~jina.Flow.to_kubernetes_yaml` is a helper function to get your stared off the ground. **There are meant to be updated and adapted to every use case**
+```
+````{admonition} Matching jina versions
+:class: caution
+If you change the Docker images for {class}`~jina.Executor` and {class}`~jina.Gateway` in your Kubernetes generated file, ensure that al of them are built with the same Jina version to guarantee compatibility.
+````
+
+You can't add basic Kubernetes feature like `Secrets`, `ConfigMap` or `Labels` via the Pythonic or YAML interface. This is intentional
+and doesn't mean that we don't support these features. On the contrary, we let you fully express your Kubernetes configuration by using the Kubernetes API to add you own Kubernetes standard to Jina.
+
+````{admonition} Hint
+:class: hint
+We recommend dumping the Kubernetes configuration files and then editing the files to suit your needs.
+````
+
+Here are possible configuration options you may need to add or change
+
+- Add labels `selector`s to the Deployments to suit your case
+- Add `requests` and `limits` for the resources of the different Pods 
+- Set up persistent volume storage to save your data on disk
+- Pass custom configuration to your Executor with `ConfigMap` 
+- Manage credentials of your Executor with secrets
+- Edit the default rolling update configuration
+
+
+(service-mesh-k8s)=
+## Required Service Mesh 
+
+```{caution}
+A Service Mesh is required to be installed and correctly configured in the K8s clusteri in which your deployed your Flow.
+```
+
+Service meshes work by attaching a tiny proxy to each of your Kubernetes pods, allowing for smart rerouting, load balancing,
+request retrying, and host of other [features](https://linkerd.io/2.11/features/).
+
+Jina relies on a service mesh to load balance request between replicas of the same Executor.
+You can use your favourite Kubernetes service mesh in combination with your Jina Flow, but the configuration files
```

Comment:
in https://docs-make-kubernetes-more-prominent--jina-docs.netlify.app/how-to/scale-out/#scale-out
Jina Flow is written as Jina `Flow`. Which formatting is correct?

Reply:
See here https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#documentation-guidelines

We do the difference when talking about concept (Flow) and code `Flow`